### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by snouatin-datastorm**
Tests unitaires et d'intégration des différents sous modules avec un taux de couverture moyen de 90%

### CODE REVIEW

File: tests/graph/test_shape_np.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        sum_shapes([])
+
+    # error if list contains non-GraphShape
+    with pytest.raises(ValueError):
+        max_shape([gs1, "not a graphshape"])
+
+
+def test_array_values_and_ordering_consistency():
+    # Ensure that array stacks edges in whatever dict insertion order is used (we rely on .values())
+    e = get_fixed_edge()
```
Comment (geofjamg) [inline tests/graph/test_shape_np.py]: not used ?

File: tests/problem/test_problem.py [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+import numpy as np
+import jax.numpy as jnp
+import pytest
+
+from energnn.problem.problem import Problem
+
+
+# Minimal Graph/Edge
+class DummyEdge:
```
Comment (geofjamg) [inline tests/problem/test_problem.py]: Why not using dummy classes ? As theses DummyEdge and DummyGraph do not match Problem types we get a few warnings.

File: tests/gnn/integration/test_coupler.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    shuffle=False,
+)
+pb_batch = next(iter(pb_loader))
+context_batch, _ = pb_batch.get_context()
+jax_context_batch = JaxGraph.from_numpy_graph(context_batch)
+context = separate_graphs(context_batch)[0]
+jax_context = JaxGraph.from_numpy_graph(context)
+
+
+class DummyCoupling:
```
Comment (geofjamg) [inline tests/gnn/integration/test_coupler.py]: Here too we get some typing warning. What would be the cleaner way to solve this ? CouplingFunction should be an asbtract class with its implementation ? So that we can also implement a test coupling function ?

File: .github/workflows/python-package.yml [hunk 4]
```diff
@@ -47,7 +47,7 @@ jobs:
       run: uv run flake8 src/energnn --count --show-source --max-complexity=10 --max-line-length=127 --statistics
 
     - name: Test with pytest
-      run: uv run pytest tests
```
Comment (geofjamg) [inline .github/workflows/python-package.yml]: Maybe for later: the build should fail if a PR decrease the global coverage

File: tests/graph/test_graph_np.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    # labels length equals number of addresses
+    assert labels.shape[0] == 3
+    # ensure that labels for 0 and 1 are equal and different from 2
+    assert labels[0] == labels[1]
+    assert labels[2] != labels[0]
+
+
+def test_offset_addresses_affects_edges_but_not_registry():
+    g1 = make_graph_with_registry(n_addresses=4, n_obj=2)
+    g2 = make_graph_with_registry(n_addresses=3, n_obj=2)
```
[the remark this answers is not in the document]
Comment (snouatin-datastorm) [reply tests/graph/test_graph_np.py]: You're right

File: tests/normalizer/test_center_reduce_function.py [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+    np.testing.assert_allclose(np.array(aux[1]), np.array(b))
+
+
+def test_compute_params_mean_std_matches_numpy():
+    cr = CenterReduceFunction()
+    # three rows, two features
+    a = jnp.array([[1.0, 2.0], [3.0, 4.0]])
+    b = jnp.array([[5.0, -1.0]])
+    aux = [a, b]
+    params = cr.compute_params(None, aux)
```
[the remark this answers is not in the document]
Comment (snouatin-datastorm) [reply tests/normalizer/test_center_reduce_function.py]: I don't see any warning due to the None while running the test. Just some Depreciationwarning

image

They remain when I pass empty array. I perhaps missed something.
